src/modules/auth/validation/authValidations.ts

```diff
@@ -0,0 +1,76 @@
+import Joi from "joi";
+
+interface User {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+    phone: number;
+    profilePicture?: string;
+    gender: "male" | "female" | "other";
+    birthDate: string;
+    language: string;
+    currency: string;
+    role: "buyer" | "seller" | "admin";
+}
+
+const authSchema = Joi.object<User>({
+    firstName: Joi.string().required().messages({
+        "string.base": "First name should be a type of text",
+        "string.empty": "First name cannot be an empty field",
+        "any.required": "First name is required"
+    }),
+    lastName: Joi.string().required().messages({
+        "string.base": "Last name should be a type of text",
+        "string.empty": "Last name cannot be an empty field",
+        "any.required": "Last name is required"
+    }),
+    email: Joi.string().email().required().messages({
+        "string.base": "Email should be a type of text",
+        "string.email": "Email must be a valid email",
+        "string.empty": "Email cannot be an empty field",
+        "any.required": "Email is required"
+    }),
+    password: Joi.string().min(8).pattern(new RegExp("^[a-zA-Z0-9]{3,30}$")).required().messages({

```

Comment:
Password validation should accept Letters, Numbers and Special Characters. For now it's not accepting special characters.

Reply:
Fixed